cfme/tests/storage/test_volume.py

```diff
@@ -20,19 +20,35 @@
 
 
 @pytest.mark.uncollectif(lambda provider: not provider.one_of(OpenStackProvider))
-def test_volume_navigation(openstack_provider):
+def test_volume_navigation(openstack_provider, appliance):
     # grab a volume name, the table returns a generator so use next
-    view = navigate_to(Volume, 'All')
+
+    collection = VolumeCollection(appliance=appliance)
+    view = navigate_to(collection, 'All')
+
     try:
-        volume_name = view.entities.table[0].name.text
-    except (StopIteration, NoSuchElementException):
+        volume_name = view.entities.get_first_entity().name
+    except(StopIteration, ItemNotFound):
         pytest.skip('Skipping volume navigation for details, no volumes present')
-    volume = Volume(name=volume_name, provider=openstack_provider)
+
+    volume = collection.instantiate(name=volume_name, provider=openstack_provider)
 
     assert view.is_displayed
 
     view = navigate_to(volume, 'Details')
     assert view.is_displayed
 
-    view = navigate_to(Volume, 'Add')
+    view = navigate_to(collection, 'Add')
     assert view.is_displayed
+
+
+@pytest.mark.uncollectif(lambda provider: not provider.one_of(OpenStackProvider))
+def test_volume_collective_crud(openstack_provider, appliance):
+    collection = VolumeCollection(appliance=appliance)
+    view = navigate_to(collection, 'All')
+
+    volumes = [collection.instantiate(name=item.name, provider=openstack_provider)
+              for item in view.entities.get_all()]
+
+    if volumes:
+        collection.delete(*volumes)
```

Comment:
REQUIRED: Hmm, this should skip if no volumes are present so we know that it actually did not do the test. Otherwise we would get pass results even though it did nothing.

Reply:
@mfalesni pytest skip added for no volume present